### ISSUE TITLE
Control-flow analysis ignores expression `cfg` attributes

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#6784]
## Environment

* **IntelliJ Rust plugin version:** 0.3.141.3638-203-nightly
* **Rust toolchain version:** 1.48.0 (7eac88abb 2020-11-16) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.3.2 Preview Ultimate Edition (IU-203.7148.15)
* **Operating system:** macOS 10.16
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Currently, our control-flow analysis does not take into account [expression `cfg` attributes](https://doc.rust-lang.org/reference/expressions.html#expression-attributes). This leads to false-positives and false-negatives in several inspections, e.g. in liveness
```Rust
fn main() {
    let a = 1; // false-negative: `a` should be annotated as unused
    #[cfg(disabled_option)] println!("{}", a);
}
```
and in move analysis
```Rust
struct S;
fn consume(s: S) {}

fn main() {
    let s = S;
    #[cfg(disabled_option)] consume(s);
    let t = s; // false-positive: Use of moved value
}
```
